README.md

```diff
@@ -23,23 +23,34 @@ Connecting those modules with each other lets you create immensly useful agents.
 
 ## Installation
 
-Beehive requires Go 1.8 or higher. Make sure you have a working Go environment. See the [install instructions](http://golang.org/doc/install.html).
+Beehive requires Go 1.8 or higher. Make sure you have a working Go environment.
+See the [installation instructions of Go](http://golang.org/doc/install.html).
 
 ### From source
 
-The recommended way is to fetch the sources and run make. This requires [dep](https://github.com/golang/dep)
-to be installed on your system.
+The recommended way is to fetch the sources and run make.
+This requires [dep](https://github.com/golang/dep) to be installed on your system.
 
+If the `$GOPATH` environment variable is empty (`echo $GOPATH`), set it with:
+    
+    # Set the $GOPATH environment variable for the active shell
+    export GOPATH="/your/go/path"
+
+Create the directory tree `$GOPATH/src/github.com/muesli/` and run this script:
+
+    cd $GOPATH/src/github.com/muesli/
+    # GNU Git should download the sources in the beehive/ directory
```

Comment:
I find this comment to be a bit superfluous.

Reply:
@muesli the Beehive license is AGPL, so the user can get the source from anyone without the need to use Git and potentially without having seen this repository and its Issues section.
We must consider the worst case: the end user does not know how Git works, so he must be informed that he is going to run GNU Git (in this way he can install it if he is missing) and that his execution _must_ produce a directory with the sources.

Reply:
Sorry, I really don't want extra comments in the commands section here. If you really consider it that important, it should be mentioned above in the "regular" README text.

Personally I still think it's a bit silly: if you're trying to compile a piece of software yourself and haven't stumbled upon git yet, you're in for a ride. We don't explain the expected behavior of "cd" and "export" here either.

Reply:
> Sorry, I really don't want extra comments in the commands section here

@muesli a code without comments is always a badly written code.

> We don't explain the expected behavior of "cd" and "export" here either.

Of course, but you are confusing external commands with built-in shell commands and system tools.
In all systems the built-ins are documented by the shell itself and therefore the user should know them; the system tools on GNU/Linux are the GNU utilities, which you can take for granted the user knows how to use (they are present on all systems and well documented, so in one way or another the user manages to use them).
Git, however, is an external command (it is not a GNU system tool, but an optional GNU software) and therefore you can not (_and must not_) make assumptions about the fact that the user has the documentation and should know how it works (so to speak, the "man" is totally optional and only a small slice of users know how it works git in detail).
As a developer it's your job to consider the worst case: the user wants to use the software and the compilation is just a hassle to accomplish this end. If put in front of a fork in the bad case the user always chooses the solution ready to use.

**If you can not provide a ready-to-use system package (means "apt install beehive" and then it works), then you must make the installation intuitive for all users, regardless of their degree of experience.**